docs/content/reference/directives.md

```diff
@@ -22,12 +22,16 @@ enum Role {
 }
 ```
 
-When we next run go generate, gqlgen will add this directive to the DirectiveRoot 
+When we next run go generate, gqlgen will add this directive to the DirectiveRoot
 ```go
 type DirectiveRoot struct {
-	HasRole func(ctx context.Context, next graphql.Resolver, role Role) (res interface{}, err error)
+	HasRole func(ctx context.Context, obj interface{}, next graphql.Resolver, role Role) (res interface{}, err error)
 }
 ```
+*ctx*: passed from the resolver
+*obj*: the payload aggregated so far
```

Comment:
This can actually vary a bit, depending on where the directive was - https://github.com/99designs/gqlgen/issues/760